### DISCUSSION TITLE
PTZ for Pan Cam V3

### DISCUSSION THREAD

**Discussion by jplwoodward** [mrlt8/docker-wyze-bridge#866] [Q&A]
Maybe I am just having trouble understanding the instructions, but I cannot get the ptz left, right... commands to work.

This throws an error (trying to move 1 degree left): http://host:5000/api/deck-cam/rotary_degree/left

Does any have this working or understand the proper http get command to do this?

Also, this: http://host:5000/api/deck-cam/rotary_degree/-10,0 returns success but has no effect.

**Comment by spuggy1968** [comment]
Now your changing the "host" to your IP of you HA Server?   That's what I needed to do.   I was unable to use the "left", but when I used values it moved the camera.

I found using the ptz_position worked well at moving the V2, but I am just hooking up my V3 and will see how it responds!

**Comment by spuggy1968** [comment]
I was playing around with my V2 Camera... the rotary_degree function would not take any negative numbers through the API:

api/dog-cam-v2/rotary_degree/-90,0   <---- this caused an error
api/dog-cam-v2/rotary_degree/10,250  <----- this worked and my camera moved upwards.

Could not get up, down, left, right to work from API... is there some proper way to send these commands through API?

**Reply by jplwoodward** [reply]
I had my local url (192.168.1.7) in place of host.  rotary_degree returns success with my V3 but does nothing.  I will try positive movement and ptz_position

**Comment by jplwoodward** [comment]
After MUCH more experimenting with my V3, here is what I have concluded.

1. left, right... do not work (or I do not understand how to use them, though i have tried many options)
2. rotary_degree does work with both positive and negative values, though sometimes the camera ignores the command.  (I think this is a Wyze issue, not a WyzeBridge issue, because it happens sometimes in the Wyze app).
3. ptz_position getting does not work, and the documentation is wrong.  If I use" ptz_position/60,135" to position my camera, then use "ptz_positon" to read the position, it reports 40,90.  If fact, it always reports the same wrong position, regardless of how the camera is pointed.  The documentation says "vertical: Degrees of the angle. Can be 0-40."  But clearly it can greater than 40 on the v3, which supports 90 up and 90 down.

**Reply by mrlt8** [reply]
I believe `ptz_position` is actually returning the current cruise point and that's why you keep getting the same `40,90`..?

**Comment by mrlt8** [comment]
Some of these issues should be fixed in the latest version.

**Reply by spuggy1968** [reply]
Thank you for the great work!

**Comment by jplwoodward** [comment]
Thank you!  Great project!

**Comment by jplwoodward** [comment]
Maybe, but I did not cruising on, nor had I ever set any cruise points on that camera.  I had just adjusted it it’s position originally in the Wyze app.  - John
